.github/scripts/stop_runner_service.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set +e
+set -x
+
+# Get the service name
+RUNNER_SERVICE=$(cat "${RUNNER_WORKSPACE}/../../.service")
+echo "GitHub self-hosted runner service: ${RUNNER_SERVICE}"
+
+if [[ -n "${RUNNER_SERVICE}" ]]; then
+  echo "The self-hosted runner has encountered an unrecoverable error and will be shutdown"
+  # and stop it to prevent the runner from receiving new jobs
+  sudo systemctl stop "${RUNNER_SERVICE}"
```

Comment:
This step will mark the job "RED" anyways since this is still in the middle of running an action, the github side would expect a response. Shutting down will mark the job "cancelled" or similar category etc. If the intention is to get rid of this runner, I think it is ok to also issue shutdown command here. 
Question is: is there a way to track such event, i.e. how frequently do we see this and what instances had the issues.

Reply:
Also, for non-ephemeral runners, systemctl stop and even shutdown would not be enough. Runner registration removal (./config.sh --remove) would be needed. 
e.g. with GCP runners, currently even if you power it down, the auto-scaler would bring it back up and sudo systemctl start after online. 
If we focus on AWS, the aws auto-scaler would need to know such event and destroy this instance, unless it destroys instances for each and every run, for this I'm not sure.

Reply:
+1 for this since stopping the service does not do node registration removal and system shutdown.

Reply:
Noted. Let me add the node registration removal and shutdown the runner completely.

> This step will mark the job "RED" anyways since this is still in the middle of running an action, the github side would expect a response

The job has already failed when this is run, so it would be a red cancel like you mention.  We can query that status from `workflow_job` Rockset collection.  You can also see it sometime on HUD with a big red "C"

On the UX size, cancelling the job could be a boon as it would be eligible for retry automatically by retry bot.

Reply:
Add the step to uninstall runner service and also shutting it down completely.  The test runner I use `i-0f1654252aaccb9c5` has disappeared from the list of registered org runner for PyTorch

Reply:
sudo ./svc.sh uninstall does not make the runner disappear. e.g. the following is what each A100 runner executes when it is online: 
"sudo ./svc.sh stop; sudo ./svc.sh uninstall; sudo ./svc.sh install ubuntu; " 
So it may have been manually removed   

We may need to hook this part with AWS auto-scaler  (cc @jeanschmidt) so that the auto-scaler proceeds to de-register the runner.

Reply:
So, I test this by logging in to an AWS Linux runner and run this combo `sudo ./svc.sh stop && sudo ./svc.sh uninstall && sudo shutdown -P now`.  The instance will be terminated on AWS and deregistered from the list of runner.  Magic!  So I'm curious what makes that happen behind the scene :)